keps/sig-apps/sidecarcontainers.md

```diff
@@ -88,7 +88,7 @@ spec:
     command: ['do something']
   - name: sidecar
     image: sidecar-image
-    sidecar: true
+    containerLifecycle: Sidecar
```

Comment:
s/containerLifecycle/lifecycle/g

It's already nested under "containers[]"

disagree?

Reply:
@thockin yeah I would've preferred that but `lifecycle` is already a field in the container spec, where PostStart and PreStop are defined.

Reply:
Even though `lifecycle` up until now has only been used to specify lifecycle _hooks_, my opinion is that it's better to extend it with something like `type: Sidecar` or `behavior: Sidecar` instead of introducing an additional `lifecycle` field (one called `containerLifecycle`).

Reply:
I think using `lifecycle` is definitely cleaner, the struct is currently just used for lifecycle hooks, but the name is more generic than that. So unless people think it would cause confusion to add a new non hook field to it, might be the best solution.

Reply:
Adding a field under the existing `lifecycle:` struct makes sense to me. It would only need a slight rewording of the [documented purpose of the lifecycle section](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.14/#lifecycle-v1-core), which is a bit hook-specific right now.